spanner/spanner_leaderboard: TestSample failed

Note: #1387 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: c158606d8f33158632f65cfc75190b78a1cbe616
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/b5bd3fe8-d818-442c-8d2b-143853fa04e1), [Sponge](http://sponge2/b5bd3fe8-d818-442c-8d2b-143853fa04e1)
status: failed
<details><summary>Test output</summary><br><pre>    leaderboard_test.go:92: run("insertscores", "projects/golang-samples-tests/instances/golang-samples-tests/databases/test-l-golang-samples-tests-3"): spanner: code = "DeadlineExceeded", desc = "Deadline Exceeded"
    leaderboard_test.go:58: got output "insertscores failed with spanner: code = \"DeadlineExceeded\", desc = \"Deadline Exceeded\""; want it to contain "Inserted scores"
    leaderboard_test.go:93: run("insertscores", "projects/golang-samples-tests/instances/golang-samples-tests/databases/test-l-golang-samples-tests-3"): spanner: code = "DeadlineExceeded", desc = "Deadline Exceeded"
    leaderboard_test.go:58: got output "insertscores failed with spanner: code = \"DeadlineExceeded\", desc = \"Deadline Exceeded\""; want it to contain "Inserted scores"
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "</pre></details>

```
=== RUN   TestSample
=== PAUSE TestSample
=== CONT  TestSample
    leaderboard_test.go:92: run("insertscores", "projects/golang-samples-tests/instances/golang-samples-tests/databases/test-l-golang-samples-tests-3"): spanner: code = "DeadlineExceeded", desc = "Deadline Exceeded"
    leaderboard_test.go:58: got output "insertscores failed with spanner: code = \"DeadlineExceeded\", desc = \"Deadline Exceeded\""; want it to contain "Inserted scores"
    leaderboard_test.go:93: run("insertscores", "projects/golang-samples-tests/instances/golang-samples-tests/databases/test-l-golang-samples-tests-3"): spanner: code = "DeadlineExceeded", desc = "Deadline Exceeded"
    leaderboard_test.go:58: got output "insertscores failed with spanner: code = \"DeadlineExceeded\", desc = \"Deadline Exceeded\""; want it to contain "Inserted scores"
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
--- FAIL: TestSample (126.57s)
```

Another `DeadlineExceeded` error. Likely related to the `PAUSE TestSample` and `CONT TestSample`. Is there a way to prevent individual samples from being interrupted?

```
=== RUN   TestSample
=== PAUSE TestSample
=== CONT  TestSample
    leaderboard_test.go:92: run(...): spanner: code = "DeadlineExceeded", desc = "Deadline Exceeded"
    leaderboard_test.go:58: got output "insertscores failed with spanner: code = \"DeadlineExceeded\", desc = \"Deadline Exceeded\""; want it to contain "Inserted scores"
    leaderboard_test.go:93: run(...): spanner: code = "DeadlineExceeded", desc = "Deadline Exceeded"
    leaderboard_test.go:58: got output "insertscores failed with spanner: code = \"DeadlineExceeded\", desc = \"Deadline Exceeded\""; want it to contain "Inserted scores"
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
    leaderboard_test.go:58: got output ""; want it to contain "PlayerId: "
--- FAIL: TestSample (126.57s)
```

Another `DeadlineExceeded` error. Likely related to the `PAUSE TestSample` and `CONT TestSample`. Is there a way to prevent individual samples from being interrupted?

--- a/spanner/spanner_leaderboard/leaderboard_test.go
+++ b/spanner/spanner_leaderboard/leaderboard_test.go
@@ -61,7 +61,10 @@ func TestSample(t *testing.T) {
 	runCommand := func(t *testing.T, cmd string, dbName string, timespan int) string {
 		t.Helper()
 		var b bytes.Buffer
-		if err := run(context.Background(), adminClient, dataClient, &b, cmd, dbName, timespan); err != nil {
+		// Set timeout to 60s so it should avoid DeadlineExceeded error.
+		cctx, cancel := context.WithTimeout(ctx, 60000*time.Millisecond)
+		defer cancel()
+		if err := run(cctx, adminClient, dataClient, &b, cmd, dbName, timespan); err != nil {
 			t.Errorf("run(%q, %q): %v", cmd, dbName, err)
 		}
 		return b.String()

